<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>950</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CAN ID Reframe Tool</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,0">
    <item>
     <layout class="QHBoxLayout" name="tables_layout" stretch="2">
      <property name="sizeConstraint">
       <enum>QLayout::SizeConstraint::SetDefaultConstraint</enum>
      </property>
      <property name="rightMargin">
       <number>9</number>
      </property>
      <item>
       <widget class="QGroupBox" name="frames_group">
        <property name="title">
         <string>Latest Received Frames</string>
        </property>
        <layout class="QVBoxLayout" name="frames_layout">
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Policy::Maximum</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>600</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QTableWidget" name="frames_table">
           <column>
            <property name="text">
             <string>Timestamp</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>ID (Hex)</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>DLC</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Datos (Hex)</string>
            </property>
           </column>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,0,0">
      <item>
       <widget class="QGroupBox" name="connection_group">
        <property name="title">
         <string>Connection Configuration</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_connection">
         <item row="1" column="1">
          <widget class="QComboBox" name="output_channel_combo"/>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_bitrate">
           <property name="text">
            <string>Bitrate (kbps):</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QComboBox" name="bitrate_combo">
           <item>
            <property name="text">
             <string>125</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>250</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>500</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>1000</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_input_channel">
           <property name="text">
            <string>Input Channel:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_output_channel">
           <property name="text">
            <string>Output Channel:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="input_channel_combo"/>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="mapping_group">
        <property name="title">
         <string>ID Mapping</string>
        </property>
        <layout class="QVBoxLayout" name="mapping_layout">
         <item>
          <widget class="QTableWidget" name="mapping_table">
           <column>
            <property name="text">
             <string>ID Original (Hex)</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>ID Reescrito (Hex)</string>
            </property>
           </column>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="mapping_buttons_layout">
           <item>
            <spacer name="mapping_buttons_spacer">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Policy::Fixed</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="add_rule_button">
             <property name="text">
              <string>Add Rule</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="delete_rule_button">
             <property name="text">
              <string>Delete Rule</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="status_group">
        <property name="title">
         <string>Status</string>
        </property>
        <layout class="QHBoxLayout" name="status_layout">
         <item>
          <spacer name="status_spacer">
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Policy::Fixed</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>80</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="status_label">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="status_indicator">
           <property name="minimumSize">
            <size>
             <width>20</width>
             <height>20</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>20</width>
             <height>20</height>
            </size>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="start_stop_button">
           <property name="text">
            <string>Start</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="logging_group">
        <property name="title">
         <string>Logging Configuration</string>
        </property>
        <layout class="QGridLayout" name="logging_layout">
         <item row="1" column="0">
          <widget class="QLabel" name="label_log_file">
           <property name="text">
            <string>Log File:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QPushButton" name="browse_log_file_button">
           <property name="text">
            <string>Browse...</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="log_file_path_edit">
           <property name="placeholderText">
            <string>Opcional: ruta al archivo de log</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>950</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuAyuda">
    <property name="title">
     <string>&amp;Ayuda</string>
    </property>
    <addaction name="actionAcerca_de"/>
   </widget>
   <addaction name="menuAyuda"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionAcerca_de">
   <property name="text">
    <string>&amp;About...</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
